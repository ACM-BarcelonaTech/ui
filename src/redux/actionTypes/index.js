const CELLS = 'CELLS';
const LOAD_CELLS = `${CELLS}.LOAD`;

const GENE_LIST = 'GENE_LIST';
const LOAD_GENE_LIST = `${GENE_LIST}.LOAD`;
const UPDATE_GENE_LIST = `${GENE_LIST}.UPDATE`;

const DIFF_EXPR = 'DIFF_EXPR';
const LOAD_DIFF_EXPR = `${DIFF_EXPR}.LOAD`;
const UPDATE_DIFF_EXPR = `${DIFF_EXPR}.UPDATE`;

const SELECTED_GENES = 'SELECTED_GENES';

const BUILD_HEATMAP_SPEC = 'BUILD_HEATMAP_SPEC';
const UPDATE_GENE_EXPRESSION = 'UPDATE_GENE_EXPRESSION';
const UPDATE_HEATMAP_SPEC = 'UPDATE_HEATMAP_SPEC';

const UPDATE_CELL_INFO = 'UPDATE_CELL_INFO';

const SET_FOCUSED_GENE = 'SET_FOCUSED_GENE';

export {
  LOAD_CELLS,
  LOAD_GENE_LIST, UPDATE_GENE_LIST, SELECTED_GENES, BUILD_HEATMAP_SPEC,
  UPDATE_GENE_EXPRESSION, UPDATE_HEATMAP_SPEC, LOAD_DIFF_EXPR, UPDATE_DIFF_EXPR,
  UPDATE_CELL_INFO, SET_FOCUSED_GENE,
};
