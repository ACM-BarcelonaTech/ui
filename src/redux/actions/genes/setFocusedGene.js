import {
  GENES_FOCUS, GENES_UNFOCUS,
} from '../../actionTypes/genes';

const setFocusedGene = (
  experimentId, gene,
) => (dispatch) => {
  if (gene) {
    dispatch({
      type: GENES_FOCUS,
      payload: {
        experimentId,
        gene,
      },
    });
  } else {
    dispatch({
      type: GENES_UNFOCUS,
      payload: {
        experimentId,
      },
    });
  }
};

export default setFocusedGene;
