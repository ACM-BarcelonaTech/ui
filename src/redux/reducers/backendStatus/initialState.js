const initialExperimentBackendStatus = {
  loading: false,
  error: false,
  status: {
  },
};

const initialState = {};

export default initialState;

export { initialExperimentBackendStatus };
