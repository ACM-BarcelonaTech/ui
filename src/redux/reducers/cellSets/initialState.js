const initialState = {
  properties: {},
  hierarchy: [],
  selected: [],
  loading: true,
  error: false,
};

export default initialState;
