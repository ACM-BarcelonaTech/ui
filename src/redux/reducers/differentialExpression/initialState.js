const initialState = {
  properties: {
    data: [],
    comparisonGroup: {},
    comparisonType: null,
    loading: false,
    error: false,
    total: 0,
  },
  comparison: {
    type: 'within',
    advancedFilters: [],
    group: {
      between: {
        cellSet: null,
        compareWith: null,
        basis: null,
      },
      within: {
        cellSet: null,
        compareWith: null,
        basis: null,
      },
    },
    ordering: {
      orderBy: 'logFC',
      orderDirection: 'DESC',
    },
  },
};

export default initialState;
