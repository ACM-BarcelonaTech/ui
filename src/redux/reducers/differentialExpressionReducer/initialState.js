const initialState = {
  properties: {
    data: {},
    cellSets: {},
    loading: false,
    error: false,
  },
};

export default initialState;
