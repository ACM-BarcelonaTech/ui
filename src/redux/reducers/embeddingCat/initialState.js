const initialState = {
  width: 700,
  height: 550,
  pointSize: 5,
  toggleInvert: '#FFFFFF',
  masterColour: '#000000',
  umap1Domain: null,
  umap2Domain: null,
  titleText: '',
  titleSize: 20,
  titleAnchor: 'start',
  axisTitlesize: 13,
  axisTicks: 13,
  transGrid: 0,
  axesOffset: 10,
  masterFont: 'sans-serif',
  xaxisText: 'UMAP 1',
  yaxisText: 'UMAP 2',
  pointStyle: 'circle',
  pointOpa: 5,
  g1Color: 'red',
  g2mColor: 'green',
  sColor: 'blue',
  legendTextColor: '#FFFFFF',
  legendEnabled: true,
  legend: null,
  geneexpLegendloc: '',
  labelSize: 28,
  labelShow: 1,
  labelFont: 2,
  labelsEnabled: true,
  selectedClusters: [],
  testVar: null,
};

export default initialState;
