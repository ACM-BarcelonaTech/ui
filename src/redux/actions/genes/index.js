import loadGeneExpression from './loadGeneExpression';
import changeGeneSelection from './changeGeneSelection';
import setFocusedGene from './setFocusedGene';
import loadPaginatedGeneProperties from './loadPaginatedGeneProperties';

export {
  loadGeneExpression,
  changeGeneSelection,
  setFocusedGene,
  loadPaginatedGeneProperties,
};
