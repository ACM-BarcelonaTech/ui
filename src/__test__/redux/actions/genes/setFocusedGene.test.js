import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import setFocusedGene from '../../../../redux/actions/genes/setFocusedGene';
import initialState from '../../../../redux/reducers/genes/initialState';

import { GENES_FOCUS, GENES_UNFOCUS } from '../../../../redux/actionTypes/genes';

const mockStore = configureStore([thunk]);

const experimentId = '1234';

describe('setFocusedGene action', () => {
  it('Dispatches focus event when gene is specified', async () => {
    const store = mockStore(initialState);
    store.dispatch(setFocusedGene(experimentId, 'asd'));

    const firstAction = store.getActions()[0];
    expect(firstAction.type).toEqual(GENES_FOCUS);
    expect(firstAction).toMatchSnapshot();
  });

  it('Dispatches unfocus event when no is specified', async () => {
    const store = mockStore({ ...initialState, focused: 'asd' });
    store.dispatch(setFocusedGene(experimentId, undefined));

    const firstAction = store.getActions()[0];
    expect(firstAction.type).toEqual(GENES_UNFOCUS);
    expect(firstAction).toMatchSnapshot();
  });
});
