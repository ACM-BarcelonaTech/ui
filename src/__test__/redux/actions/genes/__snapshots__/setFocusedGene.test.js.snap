// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setFocusedGene action Dispatches focus event when gene is specified 1`] = `
Object {
  "payload": Object {
    "experimentId": "1234",
    "gene": "asd",
  },
  "type": "genes/focus",
}
`;

exports[`setFocusedGene action Dispatches unfocus event when no is specified 1`] = `
Object {
  "payload": Object {
    "experimentId": "1234",
  },
  "type": "genes/unfocus",
}
`;
